pkg/distributor/errors.go

```diff
@@ -239,21 +240,17 @@ func handleIngesterPushError(err error) error {
 		return errors.Wrap(err, failedPushingToIngesterMessage)
 	}
 	statusCode := stat.Code()
-	if isHTTPStatusCode(statusCode) {
+	if util.IsHTTPStatusCode(statusCode) {
 		// TODO This code is needed for backwards compatibility, since ingesters may still return
 		// errors created by httpgrpc.Errorf(). If pushErr is one of those errors, we just propagate
-		// it. This code should be removed in mimir 2.12.0.
+		// it. This code should be removed together with the removal of `-ingester.return-only-grpc-errors`.
```

Comment:
I can't be sure that this code should be removed when the flag is removed. That is a conclusion that should come from a migration plan, which we don't have yet. 

Please review this comment once you have the migration plan.

Reply:
I have removed all related TODOs.

Reply:
Thanks.